exonum-light-client/src/main/java/com/exonum/client/ExonumHttpClient.java

```diff
@@ -106,6 +115,68 @@ public String getUserAgentInfo() {
     });
   }
 
+  @Override
+  public BlockResponse getBlockByHeight(long height) {
+    HttpUrl url = urlBuilder()
+        .encodedPath(BLOCK)
+        .addQueryParameter("height", String.valueOf(height))
+        .build();
+    Request request = getRequest(url);
+
+    return blockingExecuteAndParse(request, ExplorerApiHelper::parseGetBlockResponse);
+  }
+
+  @Override
+  public BlocksResponse getBlocks(int count, boolean skipEmpty, Long heightMax,
+      boolean withBlocksTime) {
+    return doGetBlocks(count, skipEmpty, heightMax, withBlocksTime);
+  }
+
+  @Override
+  public BlocksResponse getLastBlocks(int count, boolean skipEmpty, boolean withBlocksTime) {
+    return doGetBlocks(count, skipEmpty, null, withBlocksTime);
+  }
+
+  @Override
+  public Block getLastBlock() {
+    BlocksResponse response = doGetBlocks(1, false, null, false);
+
+    return response.getBlocks()
+        .stream()
+        .findFirst()
+        .orElseThrow(() -> new AssertionError("Should never happen"));
+  }
+
+  @Override
+  public Optional<Block> getLastNotEmptyBlock() {
+    BlocksResponse response = doGetBlocks(1, true, null, false);
+
+    return response.getBlocks()
+        .stream()
+        .findFirst();
+  }
+
+  private BlocksResponse doGetBlocks(int count, boolean skipEmpty, Long heightMax,
+      boolean withBlocksTime) {
+    checkArgument(count <= MAX_BLOCKS_PER_REQUEST,
+        "Requested number of blocks was {} but maximum allowed is {}",
+        count, MAX_BLOCKS_PER_REQUEST);
+    Map<String, String> query = new HashMap<>();
+    query.put("count", String.valueOf(count));
+    query.put("skip_empty_blocks", String.valueOf(skipEmpty));
+    query.put("add_blocks_time", String.valueOf(withBlocksTime));
+    if (heightMax != null) {
+      query.put("latest", String.valueOf(heightMax));
+    }
+
+    HttpUrl.Builder httpRequest = urlBuilder().encodedPath(BLOCKS);
+    query.forEach(httpRequest::addQueryParameter);
+
+    Request request = getRequest(httpRequest.build());
+
+    return blockingExecuteAndParse(request, ExplorerApiHelper::parseGetBlocksResponse);
+  }
+
   private static Request getRequest(HttpUrl url) {
```

Comment:
What if we use shorter `get`/`post`, because I always get confused by `getRequest`: it sounds as if it returns some request (it accidentally does, but that's not its point).